api/src/test/java/uk/gov/dhsc/htbhf/claimant/entitlement/CycleEntitlementCalculatorTest.java

```diff
@@ -0,0 +1,73 @@
+package uk.gov.dhsc.htbhf.claimant.entitlement;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.junit.jupiter.MockitoExtension;
+import uk.gov.dhsc.htbhf.claimant.entity.Claimant;
+
+import java.time.LocalDate;
+import java.util.List;
+
+import static java.util.Collections.singletonList;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.catchThrowableOfType;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.BDDMockito.given;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static uk.gov.dhsc.htbhf.claimant.testsupport.ClaimantTestDataFactory.aValidClaimant;
+import static uk.gov.dhsc.htbhf.claimant.testsupport.VoucherEntitlementTestDataFactory.aValidVoucherEntitlement;
+
+@ExtendWith(MockitoExtension.class)
+class CycleEntitlementCalculatorTest {
+
+    private static final int PAYMENT_CYCLE_DURATION_IN_DAYS = 3;
+    private static final int NUMBER_OF_CALCULATION_PERIODS = 3;
+
+    private EntitlementCalculator entitlementCalculator = mock(EntitlementCalculator.class);
+
+    private CycleEntitlementCalculator cycleEntitlementCalculator
+            = new CycleEntitlementCalculator(PAYMENT_CYCLE_DURATION_IN_DAYS, NUMBER_OF_CALCULATION_PERIODS, entitlementCalculator);
+
+    @Test
+    void shouldThrowExceptionWhenDurationIsNotDivisibleByNumberOfCalculationPeriods() {
+        // 10 is not divisible by 3, so should throw an exception
+        IllegalArgumentException thrown = catchThrowableOfType(
+                () -> new CycleEntitlementCalculator(10, 3, null), IllegalArgumentException.class);
```

Comment:
I's pass the mock into here instead of null as it makes it clear that that isn't the case of the Exception. Same with the tests below too.

Reply:
Done.